--- screenie.orig	2010-06-26 01:59:57.836609325 -0600
+++ screenie	2010-06-26 02:06:11.038757714 -0600
@@ -48,8 +48,7 @@
 while :
   do
       clear
-      echo "" && echo " SCREENIE - terminal screen-session handler" 
-      echo " written by Marc O. Gloor <mgloor@fhzh.ch>" && echo ""
+      echo -e "\nSCREENIE - terminal screen-session handler\n"
     
       eval $ACTIVE_SCREENS | (
       
@@ -69,11 +68,10 @@
       read ANSW
       
       case $ANSW in
-        a|a) 
+        a|A) 
           echo -n " session name: " && read sname
-          echo -n " job: " && read pjob
           sname=$(echo $sname | tr " " "_")
-          screen -S $sname -dm $pjob ;;
+          screen -S $sname ;;
         0|Q|q) 
           echo ""; exit ;;
       esac  
