--- mysql-query-browser/source/linux/gtksourceview/gtksourceview/gtksourceview.c.orig	2011-04-18 13:24:37.470205574 -0600
+++ mysql-query-browser/source/linux/gtksourceview/gtksourceview/gtksourceview.c	2011-04-18 13:25:27.984878968 -0600
@@ -1167,7 +1167,7 @@
 
 			gtk_paint_layout (GTK_WIDGET (view)->style,
 					  win,
-					  GTK_WIDGET_STATE (view),
+					  gtk_widget_get_state (view),
 					  FALSE,
 					  NULL,
 					  GTK_WIDGET (view),
@@ -1450,7 +1450,7 @@
 
 			gtk_paint_vline (widget->style, 
 					 event->window, 
-					 GTK_WIDGET_STATE (widget), 
+					 gtk_widget_get_state (widget), 
 					 &redraw_rect, 
 					 widget,
 					 "margin", 
