Index: pidgin-2.7.2/libpurple/plugins/perl/common/Makefile.PL.in
===================================================================
--- pidgin-2.7.2.orig/libpurple/plugins/perl/common/Makefile.PL.in
+++ pidgin-2.7.2/libpurple/plugins/perl/common/Makefile.PL.in
@@ -10,6 +10,7 @@ WriteMakefile(
       (ABSTRACT_FROM    => '@srcdir@/Purple.pm', # finds $ABSTRACT
        AUTHOR           => 'Purple <http://pidgin.im/>') : ()),
     'DEFINE'            => '@DEBUG_CFLAGS@',
+    'dynamic_lib' 	 => { 'OTHERLDFLAGS' => '@LDFLAGS@' },
     'INC'               => '-I. -I@srcdir@ -I@top_srcdir@ -I@top_srcdir@/libpurple @GLIB_CFLAGS@',
     'OBJECT'            => '$(O_FILES)', # link all the C files too
 #    'OPTIMIZE'          => '-g', # For debugging
Index: pidgin-2.7.2/pidgin/plugins/perl/common/Makefile.PL.in
===================================================================
--- pidgin-2.7.2.orig/pidgin/plugins/perl/common/Makefile.PL.in
+++ pidgin-2.7.2/pidgin/plugins/perl/common/Makefile.PL.in
@@ -8,7 +8,7 @@ WriteMakefile(
     ($] >= 5.005 ? ## Add these new keywords supported since 5.005
       ('ABSTRACT_FROM' => '@srcdir@/Pidgin.pm', # finds $ABSTRACT
        'AUTHOR'        => 'Pidgin <http://pidgin.im/>') :  ()),
-    'DEFINE'        => '@DEBUG_CFLAGS@',
+    'DEFINE'        => '@DEBUG_CFLAGS@ @LDFLAGS@',
     'INC'           => '-I. -I@srcdir@ -I@top_srcdir@ -I@top_srcdir@/libpurple -I@top_srcdir@/pidgin @GTK_CFLAGS@',
     'OBJECT'        => '$(O_FILES)', # link all the C files too
     'TYPEMAPS'      => ["@top_srcdir@/libpurple/plugins/perl/common/typemap"],
